data_source/gateway.go

```diff
@@ -0,0 +1,84 @@
+package dataSource
+
+import (
+	"errors"
+
+	"github.com/NethermindEth/juno/clients"
+	"github.com/NethermindEth/juno/core"
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/NethermindEth/juno/utils"
+)
+
+type Gateway struct {
+	client *clients.GatewayClient
+}
+
+func NewGateway(n utils.Network) *Gateway {
+	return &Gateway{
+		client: clients.NewGatewayClient(n.URL()),
+	}
+}
+
+func (g *Gateway) GetBlockByNumber(blockNumber uint64) (*core.Block, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (*core.Transaction, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (g *Gateway) GetClass(classHash *felt.Felt) (*core.Class, error) {
+	return nil, errors.New("not implemented")
+}
+
+func (g *Gateway) GetStateUpdate(blockNumber uint64) (*core.StateUpdate, error) {
+	response, err := g.client.GetStateUpdate(blockNumber)
+	if err != nil {
+		return nil, err
+	}
+
+	return adaptStateUpdate(response)
```

Comment:
Since what we are getting from `g.client.GetStateUpdate(blockNumber)` is actually the state update, I think we should rename this from `response` to `stateUpdate`. 

If this had been a direct call to the feeder gateway, then the naming `response` would have been appropriate.
```suggestion
	stateUpdate, err := g.client.GetStateUpdate(blockNumber)
	if err != nil {
		return nil, err
	}

	return adaptStateUpdate(stateUpdate)
```

Reply:
We cant name everything by their types. In this context what is more important is that it is the response of gateway.

Reply:
If Go supported overloading, I would've called the function `adaptResponse`